autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+                    X.shape[0],
+                    y.shape[0]
+                )
+            )
+        return X, y
+
+    def validate_features(
+        self,
+        X: Union[pd.DataFrame, np.ndarray],
+    ) -> np.ndarray:
+        """
+        Wrapper around sklearn check_array. Translates a pandas
+        Dataframe to a valid input for sklearn.
+        """
+
+        # Pre-process dataframe to make them numerical
+        if hasattr(X, "iloc"):
+            # Pandas validation provide extra user information
+            X = self._validate_dataframe_features(X)
+
+        if scipy.sparse.issparse(X):
+            X.sort_indices()
+
+        # sklearn check array will make sure we have the
+        # correct numerical features for the array
+        # Also, a numpy array will be created
+        X = sklearn.utils.check_array(
+            X,
+            force_all_finite=False,
+            accept_sparse='csr'
+        )
+        return X
+
+    def validate_target(
+        self,
+        y: Union[pd.DataFrame, np.ndarray],
+    ) -> np.ndarray:
+        """
+        Wrapper around sklearn check_array. Translates a pandas
+        Dataframe to a valid input for sklearn.
+        """
+        # Support also Pandas object in the targets
+        if hasattr(y, "iloc"):
+            y = self._validate_dataframe_target(y)
+
+        y = np.atleast_1d(y)
+        if y.ndim == 2 and y.shape[1] == 1:
+            warnings.warn("A column-vector y was passed when a 1d array was"
+                          " expected. Will change shape via np.ravel().",
+                          sklearn.utils.DataConversionWarning, stacklevel=2)
+            y = np.ravel(y)
+
+        # sklearn check array will make sure we have the
+        # correct numerical features for the array
+        # Also, a numpy array will be created
+        y = sklearn.utils.check_array(
+            y,
+            force_all_finite=True,
+            accept_sparse='csr',
+            ensure_2d=False,
+        )
+        return y
+
+    def _validate_dataframe_features(
+        self,
+        X: pd.DataFrame,
+    ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        Interprets a Pandas
+        Uses .iloc as a safe way to deal with pandas object
+        """
+
+        # Start with the features
+
+        # Register if a column needs encoding
+        enc_columns = []
+
+        # Also, register the feature types for the estimator
+        feature_types = []
+
+        # Make sure each column is a valid type
+        for i, column in enumerate(X.columns):
+            if X[column].dtype.name in self.valid_pd_enc_dtypes:
+                enc_columns.append(i)
+                feature_types.append('categorical')
+            elif not np.issubdtype(X[column].dtype, np.number):
+                if X[column].dtype.name == 'Object':
+                    raise ValueError(
+                        "Input Column {} has invalid type object. "
```

Comment:
It could also be cast to categorical or boolean, right?

Reply:
True, thanks, I will update that in the next commit.